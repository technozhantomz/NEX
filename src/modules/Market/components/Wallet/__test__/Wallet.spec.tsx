import { cleanup, render } from "@testing-library/react";
import React from "react";

import { Wallet } from "../index";

jest.mock("next/router", () => require("next-router-mock"));

jest.mock("../hooks", () => ({
  useWallet: jest.fn(() => ({
    balances: {
      baseAmount: "10.00",
      quoteAmount: "20.00",
    },
  })),
}));

describe("Wallet component", () => {
  afterEach(cleanup);

  it("renders component correctly with props", () => {
    const { asFragment } = render(<Wallet currentPair="BTC_PPY" />);

    expect(asFragment()).toMatchSnapshot();
  });

  it("renders component with correct balance values", () => {
    const { getByText } = render(<Wallet currentPair="BTC_PPY" />);

    expect(getByText("10.00")).toBeDefined();
    expect(getByText("20.00")).toBeDefined();
  });
});
