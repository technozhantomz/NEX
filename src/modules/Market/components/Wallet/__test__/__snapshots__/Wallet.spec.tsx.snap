// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Wallet component renders component correctly with props 1`] = `
<DocumentFragment>
  <h3
    class="sc-bcXHqe jOHOIm"
  >
    Assets
  </h3>
  <div
    class="sc-gswNZR kEhQcI"
  >
    <button
      class="ant-btn ant-btn-primary Button-sc-d40bd577-0 sc-dkrFOg dPgDWt fhfkqL"
      type="button"
    >
      <span>
        Deposit
      </span>
    </button>
    <button
      class="ant-btn ant-btn-primary Button-sc-d40bd577-0 sc-dkrFOg dPgDWt fhfkqL"
      type="button"
    >
      <span>
        Withdraw
      </span>
    </button>
  </div>
  <div
    class="sc-hLBbgP kBCcYD"
  >
    <span
      class="sc-jSUZER dqWKZS"
    >
      BTC Available:
    </span>
    <span
      class="sc-eDvSVe lhzfG"
    >
      20.00
    </span>
  </div>
  <div
    class="sc-hLBbgP kBCcYD"
  >
    <span
      class="sc-jSUZER dqWKZS"
    >
      PPY Available:
    </span>
    <span
      class="sc-eDvSVe lhzfG"
    >
      10.00
    </span>
  </div>
</DocumentFragment>
`;
